core/backend/src/StashManager.ts

```diff
@@ -0,0 +1,458 @@
+import { DbResult, GuidString, Id64Array, Id64String, IModelStatus, Logger, OpenMode } from "@itwin/core-bentley";
+import { ChangesetIdWithIndex, IModelError, LocalDirName, LockState } from "@itwin/core-common";
+import { existsSync, mkdirSync, readdirSync, statSync, unlinkSync } from "node:fs";
+import * as path from "path";
+import { BriefcaseManager } from "./BriefcaseManager";
+import { BriefcaseDb } from "./IModelDb";
+import { _elementWasCreated, _getHubAccess, _hubAccess, _nativeDb } from "./internal/Symbols";
+import { SQLiteDb } from "./SQLiteDb";
+import { TxnProps } from "./TxnManager";
+import { IModelHost } from "./IModelHost";
+import { BackendLoggerCategory } from "./BackendLoggerCategory";
+
+const loggerCategory = BackendLoggerCategory.StashManager;
+
+/**
+ * Custom error class for stash-related errors.
+ * @internal
+ */
+export class StashError extends Error {
```

Comment:
Do these all get caught internally and then converted to something like IModelError? I seems iffy to have a new direct Error subclass.

Reply:
Yeah; you use the bad arg status in other places.

Reply:
@khanaffan I would resolve this based on you handling my original comment, but I'm not sure about the last comment in this thread from @aruniverse.